app/models/message.rb

```diff
@@ -37,6 +37,10 @@ def self.create_from_queue(body, import = false)
       message.origin.queue_messages &&
         message.body = body['message']
 
+      if body['message'].present? and body['message']['ip'].present?
```

Comment:
So, with single line conditionals can you restructure them to be the same as the line above. i.e.

```
body['message'].present? &&
  message.ip_address = body['message']['ip']
```

I don't think you need to check for the presence of IP because if it doesn't exist, it'll just be `nil` anyway.